lib/amazonka-core/src/Amazonka/Sign/V4/Base.hs

```diff
@@ -30,7 +30,6 @@ data V4 = V4
     metaPath :: Path,
     metaEndpoint :: Endpoint,
     metaCredential :: Credential,
-    metaCanonicalPath :: CanonicalPath,
```

Comment:
I'm not too fussed about adding this to the V4 record but I could go either way. Looking at the uses, it becomes completely opaque outside the Base module due to https://github.com/mwu/amazonka/blob/no-meta-canonical-path/lib/amazonka-core/src/Amazonka/Types.hs#L422 Once returned from `base`, the only thing possible/ever done with V4 is dumping it out to logs.